source/lac/solver_control.cc

```diff
@@ -63,8 +63,6 @@ SolverControl::check (const unsigned int step,
   if (step==0)
     {
       initial_val = check_value;
-      if (history_data_enabled)
-        history_data.resize(maxsteps);
```

Comment:
It really makes no difference performancewise here because the operation is so cheap, but the right thing would probably have been to call `history_data.reserve (...)` to avoid the repeated re-allocations upon `push_back`.